pkg/config/validator.go

```diff
@@ -72,19 +73,35 @@ type (
 	// AdapterToAspectMapper returns the set of aspect kinds implemented by
 	// the given builder.
 	AdapterToAspectMapper func(builder string) KindSet
+
+	// BuilderInfoFinder is used to find specific handlers BuilderInfo for configuration.
+	BuilderInfoFinder func(name string) (*adapter.BuilderInfo, bool)
+
+	// SetupHandlerFn is used to configure handler implementation with Types associated with all the templates that
+	// it supports.
+	SetupHandlerFn func(actions []*pb.Action, constructors map[string]*pb.Constructor,
+		handlers map[string]*HandlerBuilderInfo, tmplRepo template.Repository, expr expr.TypeChecker, df expr.AttributeDescriptorFinder) error
 )
 
 // newValidator returns a validator given component validators.
 func newValidator(managerFinder AspectValidatorFinder, adapterFinder BuilderValidatorFinder,
+	builderInfoFinder BuilderInfoFinder, setupHandlerFn SetupHandlerFn, templateRepo template.Repository,
 	findAspects AdapterToAspectMapper, strict bool, typeChecker expr.TypeChecker) *validator {
 	return &validator{
-		managerFinder: managerFinder,
-		adapterFinder: adapterFinder,
-		findAspects:   findAspects,
-		strict:        strict,
-		typeChecker:   typeChecker,
+		managerFinder:   managerFinder,
```

Comment:
This struct is getting to be ridiculous, as is this entire file. We need to refactor it. We should introduce a `pkg/config/validation` package, split out the adapter, aspect, and handler logic into their own files. Then what remains of this file can just be the driver that dispatches to those other pieces.